open-api/rest-catalog-open-api.yaml

```diff
@@ -2893,6 +3003,37 @@ components:
           additionalProperties:
             type: string
 
+    AppendDataFileRequest:
+      type: object
+      required:
+        - data-files
+      properties:
+        identifier:
+          allOf:
+            - $ref: '#/components/schemas/TableIdentifier'
+          description: Table identifier to update; must be present for CommitTransactionRequest
+        accept-delay-ms:
```

Comment:
I don't feel like it's reasonable for a client to impose an SLA on a server.

Reply:
I agree, I think a clients specifying ends up leading to even more complexity since questions like @dramaticlly asked [here](https://github.com/apache/iceberg/pull/10202/files#r1589712625) naturally come up.

I think it makes sense just to exclude this. Catalog implementations can define their own SLAs and clients themselves can take whatever action they want if a certain amount of time has passed and the append wasn't committed. That keeps the spec minimal and gets us out of the "what do we do if the SLA is breached" question since that's really use case specific.

The spec would still be really useful without it since I think the main problem to solve is being able to offload the appends to services which have full visibility on ongoing appends and can run operations efficiently to minimize contention and ideally maximize throughput.